template.go

```diff
@@ -54,7 +54,8 @@ DESCRIPTION:
 
 OPTIONS:{{range .VisibleFlagCategories}}
    {{if .Name}}{{.Name}}
-   {{end}}{{range .Flags}}{{.}}{{end}}{{end}}{{else}}{{if .VisibleFlags}}
+   {{end}}{{range .Flags}}{{.}}
```

Comment:
This line results an empty line with three spaces, and two empty lines if there are any options, which breaks tests.

I'm thinking of modifying this template like this:

```diff
 OPTIONS:{{range .VisibleFlagCategories}}
    {{if .Name}}{{.Name}}
-   {{end}}{{range .Flags}}{{.}}{{end}}{{end}}{{else}}{{if .VisibleFlags}}
+   {{end}}{{range .Flags}}{{.}}
+{{else}}
+{{end}}{{end}}{{else}}{{if .VisibleFlags}}
 
 OPTIONS:
-   {{range .VisibleFlags}}{{.}}{{end}}{{end}}{{end}}
+   {{range .VisibleFlags}}{{.}}
+{{else}}
+{{end}}{{end}}{{end -}}
 `
```

This will get rid of the last empty line, and inserts one if there are no flags in the list.

Reply:
@julian7 Do you run make test after this change ? Since it impacts the unit tests those need to be updated as welll.

Reply:
I did something (which is very similar, probably the same as here), and I ran go test on them until they came back 100% successful. Then, I just copied the git diff portion here :)

This is what I have now:

```text
js@hutton in cli ☸️  yolo on  main [!] via 🐹 v1.19.1 
❯ git show | cat
commit 8335f5435f27584829776f1a34289f0da9a0a0d2
Merge: 6491dde 15b2789
Author: dearchap <dear.chap@gmail.com>
Date:   Wed Oct 5 08:33:35 2022 -0400

    Merge pull request #1513 from abitrolly/patch-1
    
    wrap: Avoid trailing whitespace for empty lines

js@hutton in cli ☸️  yolo on  main [!] via 🐹 v1.19.1
❯ git diff | cat
diff --git help_test.go help_test.go
index d7cdfa2..b0bdb7e 100644
--- help_test.go
+++ help_test.go
@@ -1370,7 +1370,7 @@ OPTIONS:
 `
 
        if output.String() != expected {
-               t.Errorf("Unexpected wrapping, got:\n%s\nexpected:\n%s",
+               t.Errorf("Unexpected wrapping, got:\n%q\nexpected:\n%q",
                        output.String(), expected)
        }
 }
@@ -1439,7 +1439,7 @@ OPTIONS:
 `
 
        if output.String() != expected {
-               t.Errorf("Unexpected wrapping, got:\n%s\nexpected: %s",
+               t.Errorf("Unexpected wrapping, got:\n%q\nexpected:\n%q",
                        output.String(), expected)
        }
 }
diff --git template.go template.go
index 9e13604..142b798 100644
--- template.go
+++ template.go
@@ -54,10 +54,14 @@ DESCRIPTION:
 
 OPTIONS:{{range .VisibleFlagCategories}}
    {{if .Name}}{{.Name}}
-   {{end}}{{range .Flags}}{{.}}{{end}}{{end}}{{else}}{{if .VisibleFlags}}
+   {{end}}{{range .Flags}}{{.}}
+{{else}}
+{{end}}{{end}}{{else}}{{if .VisibleFlags}}
 
 OPTIONS:
-   {{range .VisibleFlags}}{{.}}{{end}}{{end}}{{end}}
+   {{range .VisibleFlags}}{{.}}
+{{else}}
+{{end}}{{end}}{{end -}}
 `
 
 // SubcommandHelpTemplate is the text template for the subcommand help topic.
```